includes/Embed_Base.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * Class Embed_Block.
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories;
+
+use Google\Web_Stories\Model\Story;
+use Google\Web_Stories\Story_Renderer\Image;
+use Google\Web_Stories\Story_Renderer\Embed;
+use Google\Web_Stories\Traits\Assets;
+
+/**
+ * Embed block class.
+ */
+class Embed_Base {
+	use Assets;
+
+	/**
+	 * Player script handle.
+	 *
+	 * @var string
+	 */
+	const STORY_PLAYER_HANDLE = 'standalone-amp-story-player';
+
+	/**
+	 * Script handle.
+	 *
+	 * @var string
+	 */
+	const SCRIPT_HANDLE = 'web-stories-embed';
+
+
+	/**
+	 * Initializes the Web Stories embed block.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @return void
+	 */
+	public function init() {
+		wp_register_script( self::STORY_PLAYER_HANDLE, 'https://cdn.ampproject.org/amp-story-player-v0.js', [], 'v0', false );
+		wp_register_style( self::STORY_PLAYER_HANDLE, 'https://cdn.ampproject.org/amp-story-player-v0.css', [], 'v0' );
+
+		$this->register_script( self::SCRIPT_HANDLE, [ self::STORY_PLAYER_HANDLE, Tracking::SCRIPT_HANDLE ] );
+		$this->register_style( self::SCRIPT_HANDLE, [ self::STORY_PLAYER_HANDLE ] );
+
+		wp_localize_script(
+			self::SCRIPT_HANDLE,
+			'webStoriesEmbedBlockSettings',
+			$this->get_script_settings()
+		);
+	}
+
+	/**
+	 * Returns script settings.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @return array Script settings.
+	 */
+	private function get_script_settings() {
+		return [
+			'publicPath' => WEBSTORIES_PLUGIN_DIR_URL . 'assets/js/',
+		];
+	}
+
+	/**
+	 * Renders the block type output for given attributes.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param array  $attributes Block attributes.
+	 * @param string $content    Block content.
+	 *
+	 * @return string Rendered block type output.
+	 */
+	public function render( array $attributes, $content ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
```

Comment:
Just thinking out loud: will there be a point where the block and the shortcode don't have the same features? At that point the `render` would be different for both I assume?

Reply:
There are small difference, which is why I implement, render_shortcode and render_block. These allow for differences.

Reply:
And if the differences become too big we get rid of this render() method?

Reply:
Yeah, but we can decide that later.